### ISSUE TITLE
terradef doesn't allow for variable length arrays

### ISSUE THREAD

**Issue by tpkessler** [renehiemstra/terratest#1]
Unit tests cannot include terra arrays. Furthermore, an array declared in `terradef` has to have constant length.

```terra
import "terratest/terratest"

terra set(x: &int)
  @x = 1
end

testenv "array" do
  terradef
    var n = 1
    -- Doesn't work
    -- var x: int[n]
    var x: int[1]
    set(&x[0])
    var y = x[0]
  end
  -- Doesn't work
  -- test x[0] == 1
  test y == 1
end
```

**Comment by tpkessler** [comment]
I don't understand why a similar code works in `test5.t`. Could it be the index macro?

**Comment by renehiemstra** [comment]
Fixed in commit e0a6f42
